transport/http/outbound.go

```diff
@@ -256,6 +257,11 @@ func (o *Outbound) call(ctx context.Context, treq *transport.Request) (*transpor
 
 	span.SetTag("http.status_code", response.StatusCode)
 
+	// Service name match validation, return yarpcerrors.CodeInternal error if not match
+	if match, resSvcName := checkServiceMatch(treq.Service, &response.Header); !match {
+		return nil, yarpcerrors.InternalErrorf("Service name not match, expected: %s, Got: %s", treq.Service, resSvcName)
```

Comment:
Just one last thing before you have my stamp!
If we have a service matching error here, we want this error to be attached to the tracing span above like we discussed.

So after we make a *successful* call to roundtrip, we need to ensure that the service header matches. If it doesn't, we should assign this error to the span.

Reply:
Sorry, I forgot that, now tracing span will contain mismatch error.